### PR TITLE
[RFC] Enable -Wconversion: if_cscope.c

### PR DESCRIPTION
**Pull request by cztchoice**

### CODE REVIEW

File: src/nvim/if_cscope.c [hunk 1]
```diff
@@ -123,11 +123,12 @@ char_u *get_cscope_name(expand_T *xp, int idx)
      * the pathname of the cscope database as argument.  Only complete
      * with connection numbers. -1 can also be used to kill all
      * connections. */
+    size_t i;
```
Comment (wandernauta) [inline src/nvim/if_cscope.c]: This `i` shadows the `i` on line 90. This will work, but may be a little confusing.

File: src/nvim/if_cscope.c [hunk 2]
```diff
@@ -357,7 +358,7 @@ void cs_print_tags(void)
  */
 int cs_connection(int num, char_u *dbpath, char_u *ppath)
 {
-  int i;
+  size_t i;
```
Comment (wandernauta) [inline src/nvim/if_cscope.c]: Would it make sense to move this to the for loop initalizer below?

File: src/nvim/if_cscope.c [hunk 3]
```diff
@@ -2073,14 +2073,13 @@ static char *cs_resolve_file(int i, char *name)
...
 {
-  short i;
   if (cs_cnt_connections() == 0)
     MSG_PUTS(_("no cscope connections\n"));
   else {
     MSG_PUTS_ATTR(
         _(" # pid    database name                       prepend path\n"),
         hl_attr(HLF_T));
-    for (i = 0; i < csinfo_size; i++) {
+    for (size_t i = 0; i < csinfo_size; i++) {
```
Comment (mseeber) [inline src/nvim/if_cscope.c]: The value of i is used in format strings later. You changed the type from signed to an unsigned integral number, but the format strings expect a signed one.

**Comment by cztchoice** [comment]
Anyone have an idea about the travis failure?

**Comment by fwalch** [comment]
@cztchoice It's probably not related, the OSX builds are a bit brittle.. restarted the job.

File: src/nvim/if_cscope.c [hunk 4]
```diff
@@ -1750,25 +1752,24 @@ static void cs_print_tags_priv(char **matches, char **cntxts, int num_matches)
  *
  * read a cscope prompt (basically, skip over the ">> ")
  */
-static int cs_read_prompt(int i)
+static int cs_read_prompt(size_t i)
 {
-  int ch;
+  char ch;
   char        *buf = NULL;   /* buffer for possible error message from cscope */
   int bufpos = 0;
```
Comment (justinmk) [inline src/nvim/if_cscope.c]: can `bufpos` be `size_t`?

**Comment by justinmk** [comment]
@cztchoice This looks good, can you rebase?

**Comment by justinmk** [comment]
Thanks, our clint check became a bit more strict (it looks at surrounding lines), would you mind fixing those? Then leave a comment.

File: src/nvim/if_cscope.c [hunk 5]
```diff
@@ -1964,8 +1964,8 @@ static void cs_release_csp(int i, int freefnpp)
 static int cs_reset(exarg_T *eap)
 {
   char        **dblist = NULL, **pplist = NULL, **fllist = NULL;
-  int i;
-  char buf[20];   /* for sprintf " (#%d)" */
+  const size_t BUF_SIZE = 20;
+  char buf[BUF_SIZE];   /* for sprintf " (#%d)" */
```
Comment (cztchoice) [inline src/nvim/if_cscope.c]: @justinmk what size should i use here?

Comment (justinmk) [inline src/nvim/if_cscope.c]: It's fine at 20.

Comment (justinmk) [inline src/nvim/if_cscope.c]: Please update the comment: `/* for sprintf " (#%zu)" */`

**Comment by cztchoice** [comment]
@justinmk done, can you review this?

File: src/nvim/if_cscope.c [hunk 6]
```diff
@@ -2072,22 +2070,21 @@ static char *cs_resolve_file(int i, char *name)
...
       if (csinfo[i].fname == NULL)
         continue;
 
       if (csinfo[i].ppath != NULL)
-        (void)smsg("%2d %-5ld  %-34s  %-32s",
+        (void)smsg("%2zu %-5ld  %-34s  %-32s",
             i, (long)csinfo[i].pid, csinfo[i].fname, csinfo[i].ppath);
       else
-        (void)smsg("%2d %-5ld  %-34s  <none>",
+        (void)smsg("%2zu %-5ld  %-34s  <none>",
```
Comment (justinmk) [inline src/nvim/if_cscope.c]: Sorry I didn't notice this, but the `%-5ld` should be changed to `"%-5" PRId64`.

`%ld` is not portable.

File: src/nvim/if_cscope.c [hunk 7]
```diff
@@ -1963,8 +1962,7 @@ static void cs_release_csp(int i, int freefnpp)
 static int cs_reset(exarg_T *eap)
 {
   char        **dblist = NULL, **pplist = NULL, **fllist = NULL;
-  int i;
-  char buf[20];   /* for sprintf " (#%d)" */
+  char buf[20];   /* for snprintf " (#%d)" */
```
Comment (justinmk) [inline src/nvim/if_cscope.c]: Since we're rebasing again anyways, can you change this to `%zu`

File: src/nvim/if_cscope.c [hunk 8]
```diff
@@ -1991,7 +1989,7 @@ static int cs_reset(exarg_T *eap)
          * connection number in the same line as
          * "Added cscope database..."
          */
-        sprintf(buf, " (#%d)", i);
+        snprintf(buf, sizeof(buf), " (#%zu)", i);
```
Comment (justinmk) [inline src/nvim/if_cscope.c]: I don't think `sizeof` will work here. Use `ARRAY_SIZE(buf)` instead.

**Comment by cztchoice** [comment]
@justinmk the test run well on my own travis ci, but fails here, why?

File: src/nvim/if_cscope.c [hunk 9]
```diff
@@ -577,9 +578,8 @@ static int cs_cnt_connections(void)
...
 } /* cs_cnt_connections */
 
-static void 
-cs_reading_emsg (
-    int idx        /* connection index */
+static void cs_reading_emsg(
+    size_t idx        /* connection index */
 )
 {
   EMSGN(_("E262: error reading cscope connection %" PRId64), idx);
```
Comment (justinmk) [inline src/nvim/if_cscope.c]: https://github.com/neovim/neovim/pull/2227/files caught a problem here. 

And, I am pretty confused that this PR exists. Please search existing PRs before starting one!

Comment (justinmk) [inline src/nvim/if_cscope.c]: Anyways, if you can fix this issue, we will merge this one, and hopefully #2227 can be rebased on it.

**Comment by cztchoice** [comment]
I fix the issue, but I also merge the change from #2227 , Is this Okay?

**Comment by justinmk** [comment]
Great. Any chance you could keep the merged stuff in a separate commit and give credit to @glindste as the author?

**Comment by cztchoice** [comment]
That is not easy, because I have lost the history.. can we list as a co-author?

**Comment by justinmk** [comment]
Ok. 

    Helped-by: Gustaf Lindstedt <gustaflindstedt@gmail.com>

**Comment by glindste** [comment]
That's fine with me!

**Comment by cztchoice** [comment]
@glindste Thanks for not minding this.

**Comment by justinmk** [comment]
Merged. Thanks all

**Pull request closed without merging**